sherpa/models/model.py

```diff
@@ -488,6 +489,116 @@ def cache_model(cls, pars, xlo, *args, **kwargs):
     return cache_model
 
 
+# This shares a lot of the code with modelCacher1d, but
```

Comment:
Would it make sense to move both these routines to something like `sherpa/models/cache.py` and then we just re-export them from here? It wouldn't help with the fact that `modelCacher1d` isn't easily extensible, but does separate out the code a bit.